### PR TITLE
feat(data): replace framework.xxx with bkd.xxx

### PR DESCRIPTION
**Pull request by HydrogenSulfate**
from PR #1070 
1. replace framework.xxx with bkd.xxx in data files

### CODE REVIEW

File: deepxde/data/fpde.py [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
 import math
 
+from .. import backend as bkd
 import numpy as np
 
 from .pde import PDE
 from .. import config
```
Comment (lululxvi) [inline deepxde/data/fpde.py:9]:
```
from .pde import PDE
from .. import backend as bkd
from .. import config
```

File: deepxde/data/ide.py [hunk 2]
```diff
@@ -2,8 +2,9 @@
 
 from .helper import one_function
 from .pde import PDE
+from .. import backend as bkd
 from .. import config
-from ..backend import tf
+
```
Comment (lululxvi) [inline deepxde/data/ide.py:7]: Delete this empty line.

File: deepxde/data/fpde.py (hunk 1 above)
Comment (HydrogenSulfate) [reply deepxde/data/fpde.py:9]: Done

File: deepxde/data/ide.py (hunk 2 above)
Comment (HydrogenSulfate) [reply deepxde/data/ide.py]: Done

File: deepxde/data/ide.py [hunk 3]
```diff
@@ -56,14 +56,14 @@ def losses_train(self, targets, outputs, loss_fn, inputs, model, aux=None):
             f = [f]
         f = [fi[bcs_start[-1] :] for fi in f]
         losses = [
-            loss_fn(tf.zeros(tf.shape(fi), dtype=config.real(tf)), fi) for fi in f
+            loss_fn(bkd.zeros_like(fi), fi) for fi in f
```
Comment (lululxvi) [inline deepxde/data/ide.py:59]: Format the code via black.

File: deepxde/data/ide.py [hunk 4]
```diff
@@ -73,8 +73,8 @@ def losses_test(self, targets, outputs, loss_fn, inputs, model, aux=None):
         if not isinstance(f, (list, tuple)):
             f = [f]
         return [
-            loss_fn(tf.zeros(tf.shape(fi), dtype=config.real(tf)), fi) for fi in f
-        ] + [tf.constant(0, dtype=config.real(tf)) for _ in self.bcs]
+            loss_fn(bkd.zeros_like(fi), fi) for fi in f
+        ] + [bkd.constant(0, dtype=config.real(bkd.lib)) for _ in self.bcs]
```
Comment (lululxvi) [inline deepxde/data/ide.py:77]: There is no `bkd.constant`

File: deepxde/data/fpde.py [hunk 5]
```diff
@@ -120,8 +120,8 @@ def losses_test(self, targets, outputs, loss_fn, inputs, model, aux=None):
         if not isinstance(f, (list, tuple)):
             f = [f]
         return [
-            loss_fn(tf.zeros(tf.shape(fi), dtype=config.real(tf)), fi) for fi in f
-        ] + [tf.constant(0, dtype=config.real(tf)) for _ in self.bcs]
+            loss_fn(bkd.zeros_like(fi), fi) for fi in f
+        ] + [bkd.constant(0, dtype=config.real(bkd.lib)) for _ in self.bcs]
```
Comment (lululxvi) [inline deepxde/data/fpde.py:124]: There is no `bkd.constant`

File: deepxde/data/fpde.py [hunk 6]
```diff
@@ -370,7 +370,7 @@ def _init_weights(self):
             if self.disc.meshtype == "static"
             else self.dynamic_dist2npts(self.geom.diam) + 1
         )
-        w = [1]
+        w = [bkd.constant(1.0, dtype=config.real(bkd.lib)) if bkd.is_tensor(self.alpha) else 1.0]
```
Comment (lululxvi) [inline deepxde/data/fpde.py:373]: `bkd.constant`

Comment (HydrogenSulfate) [reply deepxde/data/fpde.py]: Done

File: deepxde/data/fpde.py (hunk 5 above)
Comment (HydrogenSulfate) [reply deepxde/data/fpde.py:124]: Done

File: deepxde/data/ide.py (hunk 4 above)
Comment (HydrogenSulfate) [reply deepxde/data/ide.py:73]: Done

File: deepxde/data/ide.py (hunk 3 above)
Comment (HydrogenSulfate) [reply deepxde/data/ide.py]: Done

File: deepxde/data/ide.py (hunk 4 above)
Comment (lululxvi) [reply deepxde/data/ide.py:73]: You don't need to add `bkd.constant`. You can use `as_tensor`.

Comment (HydrogenSulfate) [reply deepxde/data/ide.py]: Done

File: deepxde/data/fpde.py [hunk 7]
```diff
@@ -370,7 +370,7 @@ def _init_weights(self):
             if self.disc.meshtype == "static"
             else self.dynamic_dist2npts(self.geom.diam) + 1
         )
-        w = [1]
+        w = [bkd.as_tensor(1.0, dtype=config.real(bkd.lib))]
```
Comment (lululxvi) [inline deepxde/data/fpde.py:373]: if w[0] is a tensor, then there is no need to use `array_ops_compat.convert_to_array`, as it always converts to tensor

File: deepxde/data/fpde.py [hunk 8]
```diff
@@ -370,10 +370,10 @@ def _init_weights(self):
...
         for j in range(1, n):
             w.append(w[-1] * (j - 1 - self.alpha) / j)
-        return array_ops_compat.convert_to_array(w)
+        return w
```
Comment (lululxvi) [inline deepxde/data/fpde.py:376]: Needs bkd.concat?

**Pull request merged**